### PR TITLE
doc: configure system variables

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Let users/developers know where they can set system variables.

If this should be put into the user doc, I am also fine with it.

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]

**Comment by lmatz** [comment]
why does it ask for a license header in `src/prost/src/lib.rs`?

File: docs/developer-guide.md [hunk 1]
```diff
@@ -157,6 +157,13 @@ Now you can run `./risedev d` to start a new dev cluster. The new dev cluster wi
 
 You may also add multiple compute nodes in the cluster. The `ci-3cn-1fe` config is an example.
 
+### Configure system variables
+
+Users can configure or override system variables in `src/config/risingwave.toml`.
```
Comment (hengm3467) [inline docs/developer-guide.md:162]: For the consistency of tone, I suggest we change "Users" to "You".

File: docs/developer-guide.md [hunk 2]
```diff
@@ -157,6 +157,13 @@ Now you can run `./risedev d` to start a new dev cluster. The new dev cluster wi
 
 You may also add multiple compute nodes in the cluster. The `ci-3cn-1fe` config is an example.
 
+### Configure system variables
+
+You can configure or override system variables in `src/config/risingwave.toml`.
+Not all the variables are explicitly set in this file. 
+You may check `src/common/src/config.rs` to see all the configurable variables,
+and set each one in the corresponding section such as `[server]`, `[storage]`.
```
Comment (hengm3467) [inline docs/developer-guide.md:165]: @lmatz Question: If developers want to include additional variables, they need to add variables and values to the .toml file. Right? Does it matter if a variable is added to a particular section vs another one? If not, we can say something like this:
"You can check ... to see all the configurable variables. If additional variables are needed, edit `src/config/risingwave.toml` to include them [in the corresponding section].

Comment (lmatz) [reply docs/developer-guide.md]: Yes, they do.
Yes, it matters.

do you mean `if yes, we can say something like this:`?

Comment (hengm3467) [reply docs/developer-guide.md]: @lmatz Yes. If yes, then here is the proposed wording:
You can check ... to see all the configurable variables. If additional variables are needed, include them in the correct sections (such as `[server]` or `[storage]') in `src/config/risingwave.toml`.

**Pull request merged**